File: ADCP_Nortek_Aquadopp_AWAC_Vector/Andere talen & old/Matlab/script_mariakerke_2013_08.py
from __future__ import annotations

import numpy as np

from campaign_runner import run_campaign_entries, show


def main() -> list[dict]:
    entries = [
        {
            "dpl_param": {
                "folder_name": "../data/2013_08_mariakerke/A847102",
                "dpl_name": "A847102",
                "reference_height": "aquadopp",
                "vertical_dir": "downward",
                "time_gap": 0,
            },
            "out_param": {"save_format": "png"},
            "days_per_subplot": 7,
            "ebb_flood": False,
            "ellipse_bounding_box_display": [np.nan, np.nan, np.nan, np.nan],
        }
    ]
    return run_campaign_entries(entries)


if __name__ == "__main__":
    main()
    show()

File: Instrumenten/ADCP/ADCP_Nortek_Aquadopp_AWAC_Vector/Andere talen & old/Matlab/script_mariakerke_2013_08.m
% SCRIPT_MARIAKERKE_2013_08 is a script that generates the figures of the
% August 2013 Mariakerke campaign (2 aquadopp and 1 AWAC)

% Required packages: suplabel
%
% Developed at Flanders Hydraulics Research (FHR), Antwerp, Belgium
% Author: Olivier Gourgue

% test changement

close all
clear all

% dpl_param.data_treatment = 'storm';
out_param.save_format = 'png';

% aquadopp
dpl_param.folder_name = '../data/2013_08_mariakerke/A847102';
dpl_param.dpl_name = 'A847102';
% dpl_param.z0 = -4.5;
dpl_param.reference_height = 'aquadopp';
dpl_param.vertical_dir = 'downward';
dpl_param.time_gap = 0; % to change 
dpl = load_deployment_files(dpl_param);

out = read_output_parameters(dpl, out_param);

figure_ts_instrument_orientation(dpl, out);
figure_ts_depth_averaged_velocity_components(dpl, out);
figure_ts_velocity_norm_profile(dpl, out, 7, NaN, NaN, NaN);
figure_depth_averaged_horizontal_velocity(dpl, out, 0, NaN, [NaN NaN NaN NaN]);